openlibrary/plugins/worksearch/code.py

```diff
@@ -767,6 +775,40 @@ def GET(self):
         web.header('Content-Type', 'application/json')
         return delegate.RawText(json.dumps(response))
 
+
+@public
+def work_search(query, sort=None, page=1, offset=0, limit=100):
+    """
+    params:
+    query: dict
+    sort: str editions|old|new|scans
+    """
+    sorts = {
+        'editions': 'edition_count desc',
+        'old': 'first_publish_year asc',
+        'new': 'first_publish_year desc',
+        'scans': 'ia_count desc'
+    }
+    query['wt'] = 'json'
+
+    try:
+        (reply, solr_select, q_list) = run_solr_query(query,
+                                                      rows=limit,
+                                                      page=page,
+                                                      sort=sort and sorts[sort] or None,
```

Comment:
SUGGESTION: I assume `and` vs. `or` precedence will work out, but you might consider parentheses to keep the reader from wondering.

Reply:
Good point; this is legacy code that was moved.

I think this can probably be simplified to `sort=sorts.get(sort)` which defaults to None